feat(theming): land Ant Design v5 overhaul — dynamic themes, real dark mode + massive styling refactor

mistercrunch opened the pull request:
This merges the long-running “theming” branch into `master`.

Superset is now *fully*, *dynamically* themeable. We’ve moved to Ant Design v5’s
token-based theming and css-in-js, killed the last traces of Bootstrap/.less,
and unlocked a first-class dark mode.

### Highlights
* Full migration to Antd v5 with major a11y gains and 100 % css-in-js
* Removed Bootstrap, Font Awesome, and the old .less pipeline
* Rebuilt remaining FAB server-side CRUD views with modern React components
* Core visualizations render cleanly in both light & dark themes
* Design atoms broken out to `@superset-ui/core/components` for easy reuse
* Antd token system exposes granular per-component overrides
* Config-driven themes (`superset_config.py`); WYSIWYG editor behind a FF
* Theming works without rebuilds; OS-level auto-switch coming next

### By the numbers
* 1 267 files touched, ~50 000 LOC diff, 191 commits
* 6-month effort, 4-5 heavy contributors, rebased on `master` the whole way

### Known gaps / follow-ups
* ~400 places still use legacy `theme.colors.*` tokens
* Theme CRUD UI + assignment logic (per-dashboard, per-user) still TBD
* CSS templates still load, but some may need tweaks for the new tokens
* Full visual QA pass is in flight before we cut the **6.0 “Theming Release”**

### BREAKING CHANGES
* `THEME_OVERRIDES` is gone — configs must move to the new Antd-compatible JSON
* Any custom CSS templates relying on the old token names may need updates

Huge thanks to everyone who battled merge conflicts and kept this branch alive.
Ready for prime time — let’s ship it. 🚀

mistercrunch committed:
feat: messing with the theme

mistercrunch commented on .pre-commit-config.yaml:
```diff
@@ -68,6 +60,14 @@ repos:
           - prettier@3.3.3
         args: ["--ignore-path=./superset-frontend/.prettierignore"]
         files: "superset-frontend"
+  - repo: local
```
Not directly related to PR, but makes more sense to run prettier first as it's in "fix" more and will often silence double-error-logging between itself and eslint.

mistercrunch commented on superset-frontend/packages/generator-superset/generators/plugin-chart/templates/src/MyChart.erb:
```diff
@@ -25,29 +25,14 @@ import { <%= packageLabel %>Props, <%= packageLabel %>StylesProps } from './type
   border-radius: ${({ theme }) => theme.gridUnit * 2}px;
   height: ${({ height }) => height}px;
   width: ${({ width }) => width}px;
-
```
removing this from plugin template as really "you should use CSS minimally in your plugins"

mistercrunch commented:
/korbit-review

mistercrunch commented on superset-frontend/packages/superset-ui-chart-controls/src/components/ColumnTypeLabel/ColumnTypeLabel.tsx:
```diff
@@ -37,9 +37,9 @@ const TypeIconWrapper = styled.div`
-    width: ${theme.gridUnit * 6}px;
-    height: ${theme.gridUnit * 6}px;
-    margin-right: ${theme.gridUnit}px;
+    width: ${theme.sizeUnit * 6}px;
+    height: ${theme.sizeUnit * 6}px;
```
Right, for now `sizeUnit === gridUnit === 4`, so just aligning with AntD's naming here. I believe most padding/margin are a factor of this (or arguably should be if they aren't). Guessing the `compactAlgorithm()` in AntD mostly just scales down this one token and things like font sizes down.

mistercrunch commented on superset-frontend/packages/superset-ui-core/src/theme/types.ts:
```diff
@@ -0,0 +1,334 @@
+ * This array is used to define which keys from the full Antd token set
+ * we actually allow in the SupersetTheme.
+ */
+export const allowedAntdTokens = [
```
### about the token list
About allowing/denying certain tokens, I started with a deny list of patterns, mostly filtering out they're "system colors" (https://ant.design/docs/spec/colors#system-level-color-system) as I wanted only the "functional colors" to flow through. We do have needs around data viz palettes but felt like functional colors were not the right thing for us. I really polluted the token namespace with lots of things I'd argue that we shouldn't use in Superset.

I'd say we really need a flood gate here, and though being explicit about it was right, and proper ts types are really needed here, so that the compiler can tell you if you're imagining tokens or not.

I'd say we could be more thoughtful about limiting this list, but didn't want to run a process to classify all tokens (useful/not-useful) as it's pretty subjective and in many ways AntD's team has done that work already.

# about sizing/spacing
Unclear what we want to do here, but allowing for a more `compact` theme would be achieved through these tokens. It would be interesting to apply AntD's compactAlgorightm and what it looks like for Superset. Haven't done it yet but should be super easy to just apply it and take a tour of the app.

michael-s-molina commented:
@mistercrunch I'm a little confused. Is this PR ready for review or are you still moving things to separate PRs?

> [factoring out pre-commit changes into another PR](https://github.com/apache/superset/pull/31590/commits/ef35198e4855f9c72022b1ae0a01ea5469ebc838)
[fix(Menu): Adjust hover and bottom styles in light mode](https://github.com/apache/superset/pull/31590/commits/8133d480dd24da5edd698c4d6574bc55f4bf0c61)

If the work is still in progress, can you convert this PR to a draft?

michael-s-molina commented:
@mistercrunch I pulled this branch and did a quick navigation. I understand the importance of merging this before the breaking window closes and that we can fix the issues during the RC rounds but I think we need to polish a little bit more before merging this PR. Here's some examples of things it would be good to fix before merging this PR:

1. The home page style looks off and there's a weird toggle at the top. I believe this is used to change from light to dark theme and it should be hidden.
<img width="1911" alt="Screenshot 2025-01-29 at 16 02 43" src="https://github.com/user-attachments/assets/e58515c2-34c4-495c-bf07-094ccdc3c815" />

2. Tabs style
<img width="1890" alt="Screenshot 2025-01-29 at 16 03 57" src="https://github.com/user-attachments/assets/ea59410a-c401-4807-9a49-9a7368de607d" />

3. Empty container icon
<img width="1907" alt="Screenshot 2025-01-29 at 16 13 24" src="https://github.com/user-attachments/assets/a83f32f9-74eb-4f89-ac7a-e3f108326ed3" />

4. Icons size
<img width="1910" alt="Screenshot 2025-01-29 at 16 04 41" src="https://github.com/user-attachments/assets/1a2f2c8e-b077-4fc5-b3d3-09c054c0e66e" />

5. Controls, drag-and-drop image, buttons
<img width="1908" alt="Screenshot 2025-01-29 at 16 05 19" src="https://github.com/user-attachments/assets/bc1f9a60-71ef-46d2-aa41-24853714ca55" />

6. Font sizes for the error messages
<img width="1899" alt="Screenshot 2025-01-29 at 16 06 37" src="https://github.com/user-attachments/assets/80ae89a8-ad7d-42e0-8c24-1ec098be3c86" />

7. Menu buttons
<img width="305" alt="Screenshot 2025-01-29 at 16 07 01" src="https://github.com/user-attachments/assets/1b8ee54d-751c-4d3c-9f30-a71c2bcdb350" />

michael-s-molina commented:
It's also important to add a message to `UPDATING.md` explaining how this change affects folks using the [THEME_OVERRIDES](https://github.com/apache/superset/blob/732de4ac7fae88e29b7f123b6cbb2d7cd411b0e4/superset/config.py#L671) configuration and how can they migrate to the new schema.

mistercrunch commented:
It's ready for review if reviewers accept a large PR. At this point I moved the larger refactors out of here and this should be basically:
- a new theming system (see `packages/superset-ui-core/theme`)
- aligning the codebase with new theming system (mapping old tokens to new tokens)

mistercrunch commented:
Thanks for reporting the visual issues, planning on solving the bulk of them (and other ones we find) prior to merging this.

mistercrunch commented:
/testenv up

betodealmeida reviewed (approved):
Beautiful!

mistercrunch commented on superset-frontend/.storybook/preview.jsx:
```diff
@@ -17,8 +17,8 @@
-import { supersetTheme, ThemeProvider } from '@superset-ui/core';
-import { AntdThemeProvider } from '../src/components/AntdThemeProvider';
+import { themeObject } from '@superset-ui/core';
+import { theme } from 'src/preamble';
```
oh weird, guessing eslint might not be covering this folder...

mistercrunch commented on superset-frontend/packages/superset-ui-chart-controls/src/components/SQLPopover.tsx:
```diff
@@ -66,8 +66,8 @@ export const SQLPopover = (props: PopoverProps & { sqlExpression: string }) => {
-            background: theme.colors.secondary.light5,
-            maxWidth: theme.gridUnit * 100,
+            background: theme.colorPrimaryBg,
+            maxWidth: theme.sizeUnit * 100,
```
Yeah I think `sizeUnit` is best used for spacing (margin/padding), though the thing to understand is that a `compact` mode would typically be achieved by sizing down `sizeUnit`, so if you want your component size to grow/shrink with everything else here you might want to use it or not.

mistercrunch commented on superset-frontend/packages/superset-ui-core/src/theme/Theme.tsx:
```diff
@@ -0,0 +1,352 @@
+
+  private static readonly sizeMap: Record<FontSizeKey, string> = {
+    xs: 'fontSizeXS',
+    s: 'fontSizeSM',
```
I hear you, had similar thoughts but decided to commit to the AntD token semantics that are well documented and supported across AntD components and beyond. More on it here -> https://ant.design/docs/react/customize-theme

mistercrunch commented on superset-frontend/packages/superset-ui-core/src/theme/Theme.tsx:
```diff
@@ -0,0 +1,352 @@
+  private constructor({
+    seed,
+    antdConfig,
+    isDark = false,
```
Dark is a pretty funky construct where based on this we pass in an AntD `darkAlgorithm` object in `ThemeConfig`. Similarly they offer a `compact` algorithm that can be composed with `darkAlgorithm`, so you can do 
```
const themeConfig = {
  seed,
  algorithm: [darkAlgorithm, compactAlgorithm]
}
```
Meaning some of these things are not exclusive, which is a bit funky. 

The constructor here allows for different combos through static methods, like `Theme.fromSeed(seed, isDark)` and  another `Theme.fromAntdConfig(config)`. We still need `isDark` to support legacy logic around `theme.colors.primary.dark3`. 

Could simplify/cleanup later, and until the public API settles. Working with all this, I think standardizing with an extended AntD config makes most sense. If you want a dark theme, give me a dark antdconfig....

mistercrunch commented on superset-frontend/packages/superset-ui-core/src/theme/Theme.tsx:
```diff
@@ -0,0 +1,352 @@
+  }
+
+  mergeTheme(partialTheme: Partial<LegacySupersetTheme>): void {
+    // const mergedTheme = merge({}, this.theme, partialTheme);
+    // const isDark = tinycolor(mergedTheme.colorBgBase).isDark();
+    // const antdConfig = Theme.getAntdConfig(systemColors, isDark);
+    // this.updateTheme(mergedTheme, antdConfig, isDark);
```
mmmh, yeah that one is really a `TODO: allow people to override the theme through THEME_OVERRIDES in config.py`. Needs more thinking around how we allow people to set the theme from a config object. A bit of an artifact from previous theme handling, but it remains hooked in from bootstrap_data in `src/preamble.ts`. 

Good to resurface this as it's a pretty important hanging thread in the whole theming saga. Do we want to allow people to set the theme through config? build time? in-UI?

I'm thinking for now I might want to simply call this method `setTheme`, and pass what's in the config.py/bootstrap_data as a `seed`. Pure antd config aren't serializable as they can contain algorightm as object reference, which we can't reference from the backend. 

Needs more thinking, but I'm thinking I'll probably do something simple so that it's possible to set the theme from config.THEME_OVERRIDES

mistercrunch commented on superset-frontend/plugins/plugin-chart-echarts/src/BigNumber/BigNumberViz.tsx:
```diff
@@ -374,7 +374,7 @@ export default styled(BigNumberVis)`
       .header-line,
       .subheader-line {
-        opacity: ${theme.opacity.mediumHeavy};
+        opacity: 60%;
```
oh that was a bit of a hard slash (machete in the jungle), here's the rationale for it:
- The antd theming system doesn't specify presets for opacity
- use/reuse across the codebase was minimal, with hard-coded widespread
- is there real value around having set values? what should they be? does it help to use midHeavy over 80% or 70%. Would a "themer"  want to alter these values centrally ,like "oh I want all the mediumHeavy to be a little lighter in my environment"?

msyavuz commented on superset-frontend/packages/superset-ui-core/src/theme/Theme.tsx:
```diff
@@ -0,0 +1,352 @@
+    color: string,
+    isDark: boolean,
+  ): DeprecatedColorVariations {
+    const bg = isDark ? '#FFF' : '#000';
+    const fg = isDark ? '#000' : '#FFF';
```
Naming here seems a bit off. bg is white when isDark?

msyavuz commented on superset-frontend/packages/superset-ui-core/src/theme/index.tsx:
```diff
@@ -0,0 +1,71 @@
+
+const styled = emotionStyled;
+
+// launching in in dark mode for now while iterating
```
```suggestion
```

msyavuz commented on superset-frontend/packages/superset-ui-core/src/theme/Theme.tsx:
```diff
@@ -0,0 +1,352 @@
+  }
+}
+
+/* eslint-enable theme-colors/no-literal-colors */
```
```suggestion
```

msyavuz reviewed (approved):
There are some nits(leftover comments, naming etc.) but it looks good

msyavuz commented:
There are also some more visual bugs? on dark mode:

1. Chart text on Dashboard is black on dark mode
![image](https://github.com/user-attachments/assets/3d599d5b-ce59-49c2-985a-8052baa48da8)

2. Explore page white bg on bottom
![image](https://github.com/user-attachments/assets/3bddbdd1-b713-4593-bc25-4541180b950e)

3. Sqllab tabs have white bg
![image](https://github.com/user-attachments/assets/4d4c12d7-0af9-461c-8039-b0945101baf0)

4. Some inputs don't change in dark mode which i think is unintentional
![image](https://github.com/user-attachments/assets/62c5e3ad-f3be-49b1-839a-2354145fb9ce)

Also i think there is a dark mode Superset logo which we can use in dark mode.

mistercrunch committed:
make Cancel secondary

mistercrunch committed:
adjusting some buttons

mistercrunch committed:
fix types/tests

mistercrunch committed:
fix unit test

mistercrunch committed:
ts-ignore

mistercrunch commented:
Hey, just clarifying here, the dark mode switch (behind a feature flag) is expected to expose styling issues until we can remove all of our reliance on `.less` files, and until everything is move to css-to-jss. Big part of the puzzle is the antd-v5 migration that's well under way.

For the time being this switch helps visualize where things aren't themed properly yet. Also note that during this [hopefully short] transition period, running `npm run compile-less` should fix issues as our `.less` files are now templated with Handlebars (while transitioning).

mistercrunch committed:
fix a few buttons

kgabryje commented:
Looks much better already! I've got some second pass comments:

### Dashboard list page
Before:

<img width="1728" alt="image" src="https://github.com/user-attachments/assets/44f7502f-2c75-43fd-bc7f-22bcaccc6ce3" />

After:

<img width="1728" alt="image" src="https://github.com/user-attachments/assets/9fe94ff8-d4be-4f9c-bd2f-b0298eae13c5" />

1. Secondary button looks different
2. Buttons have larger border radius than before
3. Spacing between the header and the filters section is smaller
4. Import dashboards icon has different color
5. Filters placeholders were light grey before
6. When you click on a filter, the active option in the menu has darker color than before

### Dashboard
1. Native filters bar has gray background instead of white
2. The gray background behind the metadata bar is gone
3. Font size of the dashboard header is smaller
4. Table chart looks a bit different - now the header is gray, and before every other row had a light gray background, now its all white
5. Native filters modal - spacings are a bit off, the settings/scoping tabs background is gray
6. In edit mode, the right panel has gray background, and "create new chart" button has different color
7. Discard button is primary, should be secondary


### Explore
1. Borders is controls are much darker
2. Gray background behind Update Chart button
3. In table chart, when I applied conditional formatting, the colors are much lighter (more transparent??) than before

mistercrunch committed:
make secondary filled/primary

mistercrunch committed:
fix dashboard left panel background-color

mistercrunch committed:
fixing storybook

mistercrunch commented:
### Dashboard list page

> Secondary button looks different

Trying to address this with secondary="filled" with color="primary" instead of "outlined" / "grayscale"

> Buttons have larger border radius than before

Going vanilla! We could reduce it if we want to maintain consistency with previous theme for this specifically

> Spacing between the header and the filters section is smaller

TODO

> Import dashboards icon has different color
TODO: I'm a bit puzzled by this one 

> Filters placeholders were light grey before

Should fix itself as we migrate Form to antd-v5

> When you click on a filter, the active option in the menu has darker color than before

Should fix itself as migrate Select to antd-v5

### Dashboard
> Native filters bar has gray background instead of white

Fixed!

> The gray background behind the metadata bar is gone

Fixed, using `colorLayout` instead of colorBgElevated, which is white in the theme

> Font size of the dashboard header is smaller
looks like it's 20px now

> Table chart looks a bit different - now the header is gray, and before every other row had a light gray background, now its all white

TODO

> Native filters modal - spacings are a bit off, the settings/scoping tabs background is gray

TODO

> In edit mode, the right panel has gray background, and "create new chart" button has different color
Fixed!

> Discard button is primary, should be secondary

Fixed!

### Explore
> Borders is controls are much darker

mmmh is that for large Selects? Like the box containing the things in the select?

> Gray background behind Update Chart button
fixed! (I think...)

> In table chart, when I applied conditional formatting, the colors are much lighter (more transparent??) than before

TODO

mistercrunch commented:
Few notes:

- the FAB views look pretty off, not sure if it's a dealbreaker. To be fair, they look bad on `master` too, but worse now after this PR. Goal is to kill them, so wondering if a regression in a to-be-deprecated area is tolerable.
- reminder to turn off the dark theme switch feature flag before merging...

mistercrunch committed:
set DARK ff to false

mistercrunch commented:
about 

>- Font size of the dashboard header is smaller

I checked and the the branch is at `20px` while master is at `21px`, good eye for spotting this @kgabryje  :) . I think 20px is ok :)

mistercrunch committed:
switch primary color, keep old hex as comment in case we want to rollback

mistercrunch committed:
rebased

mistercrunch committed:
echarts

mistercrunch committed:
fix treemap

mistercrunch committed:
fix sankey theme

mistercrunch committed:
fix BIG number

mistercrunch committed:
fixing vizes

mistercrunch committed:
minor progress

mistercrunch committed:
merging

mistercrunch committed:
fix MoreOutlined button background

mistercrunch committed:
color

mistercrunch committed:
more button background

mistercrunch committed:
echarts theming

msyavuz commented:
There are two visual issues that i found while working branched out from this pr:

1. Hover state in database modal database buttons looks off

![image](https://github.com/user-attachments/assets/506dc022-74fd-48ac-a857-cb4c32d8471d)

2. Second one is introduced with recent changes. Filter settings icon color is different.

![image](https://github.com/user-attachments/assets/c137a6b0-e2ee-4116-a747-bca10bb08955)

mistercrunch commented:
Yeah found a few after the latest merge, will do further merge/fix/push today. 

@msyavuz one thing I found with the recent `<Icon />` changes is it would be nice if the component would simply inherit the color from parent. I don't think it does now.

mistercrunch committed:
Merge branch 'master' into template_less

msyavuz commented:
> @msyavuz one thing I found with the recent <Icon /> changes is it would be nice if the component would simply inherit the color from parent. I don't think it does now.
Yes, it defaults to `theme.colors.grayscale.base`.

mistercrunch committed:
some theming fixes

mistercrunch committed:
touchups

mistercrunch committed:
Merge branch 'master' into template_less

mistercrunch committed:
fix CopyOutlined icon in dataset view to align with sizing

mistercrunch committed:
theme toast icons and various cosmetic updates

mistercrunch committed:
Fix table row separator in TableCollection

mistercrunch committed:
fix icon on Explore's cached label

mistercrunch committed:
going vanila on tooltips

mistercrunch committed:
aligning some colors tokens

mistercrunch committed:
fix table viz colors

mistercrunch committed:
fix Collapse color content

mistercrunch committed:
fix theme for ag-grid in SQL LAB

mistercrunch committed:
theming react-code-highlighter or whatev it's called

mistercrunch committed:
Merge branch 'master' into template_less

mistercrunch committed:
improve styles in HighlightedSql

mistercrunch committed:
fix some tests

mistercrunch committed:
fix test

mistercrunch committed:
fix theme test

mistercrunch committed:
fix another test

mistercrunch committed:
fix types in Gauge/transformProps.test.ts

mistercrunch committed:
fix test for BasicErrorAlert.tsx

mistercrunch committed:
fix SavedQueries test

mistercrunch committed:
fix ErrorMessageWithStackTrace

mistercrunch committed:
fix test in dashboard/actions.test.js

mistercrunch committed:
add theming support to Storybook

mistercrunch committed:
fixing themes in Storybook

mistercrunch committed:
styling the AsyncAceEditor

mistercrunch committed:
minor color tweak

mistercrunch committed:
a theme editor

mistercrunch committed:
ThemeEditor

mistercrunch committed:
ThemeEditor

mistercrunch committed:
Merge branch 'master' into template_less

mistercrunch committed:
fix cypress test

mistercrunch committed:
oopsy daisy

mistercrunch committed:
one minor edit to fix a cypress test

EnxDev committed:
refactor(table): Upgrade table component from antd4 to antd5 (#32378)

Co-authored-by: Diego Pucci <diegopucci.me@gmail.com>
Co-authored-by: Mehmet Salih Yavuz <salih.yavuz@proton.me>

mistercrunch committed:
Merge branch 'master' into template_less

mistercrunch committed:
set Global body bg color to colorBgBase

mistercrunch committed:
centralize theming stuff in AsyncAceEditor

mistercrunch committed:
altering comment for THEME_OVERRIDES in superset/config.py

mistercrunch committed:
lint

mistercrunch committed:
Merge branch 'master' into template_less

msyavuz committed:
refactor(form): Migrate Form component to Ant Design 5 (#32729)

mistercrunch committed:
force colorLink in GlobalStyles

mistercrunch committed:
fix links and bring GlobalStyles to superset-ui/core

mistercrunch committed:
remove console.log

mistercrunch committed:
theming the viz picker

mistercrunch committed:
fix some type issues

mistercrunch committed:
feat: Theme to include brand logo configuration

msyavuz committed:
refactor(theming): Fixes to previously migrated components (#32845)

mistercrunch committed:
Merge branch 'master' into template_less

mistercrunch committed:
fix merge-related issues

EnxDev committed:
refactor(Components): Create wrappers for Ant Design 5 direct exports (#32705)

Co-authored-by: Diego Pucci <diegopucci.me@gmail.com>
Co-authored-by: Geido <60598000+geido@users.noreply.github.com>
Co-authored-by: Mehmet Salih Yavuz <salih.yavuz@proton.me>

alexandrusoare committed:
refactor(Tabs): Upgrade Tabs to Antd5 (#32810)

Co-authored-by: Maxime Beauchemin <maximebeauchemin@gmail.com>

mistercrunch committed:
Merge branch 'master' into template_less

DamianPendrak committed:
refactor(Collapse): Upgrade Collapse to Antd5 (#32959)

Co-authored-by: Maxime Beauchemin <maximebeauchemin@gmail.com>

mistercrunch committed:
use colorBorder in AddSliceCard

msyavuz committed:
refactor(Select): Migrate Select component to Ant Design 5 (#32514)

Co-authored-by: Maxime Beauchemin <maximebeauchemin@gmail.com>
Co-authored-by: Diego Pucci <diegopucci.me@gmail.com>

mistercrunch committed:
Merge branch 'master' into template_less

mistercrunch committed:
Fix theming in Explore west panel - Search Input

msyavuz committed:
refactor(Table): Use our custom Table component in CRUD views (#32964)

Co-authored-by: Geido <60598000+geido@users.noreply.github.com>

mistercrunch committed:
restyling page header and buttons

mistercrunch committed:
fix: use Input & TextArea instead of native <input> and <textarea> (#32989)

payose committed:
refactor(Checkbox): Upgrade component to Ant Design v5 (#32980)

Co-authored-by: Diego Pucci <diegopucci.me@gmail.com>

mistercrunch committed:
Merge branch 'master' into template_less

EnxDev committed:
refactor(breadcrumb): Upgrade Breadcrumb component from AntD v4 to AntD v5  (#32905)

geido committed:
refactor(Components): Use named imports - P.1 (#33081)

mistercrunch committed:
Merge branch 'master' into template_less

mistercrunch committed:
fix a python build issue

mistercrunch committed:
fix scripts/change_detector.py

mistercrunch committed:
fix a unit test

EnxDev committed:
Merge branch 'master' into template_less

mistercrunch committed:
Merge branch 'master' into template_less

mistercrunch committed:
post merge conflicts

mistercrunch committed:
fix imports

msyavuz committed:
fix: provide id's

msyavuz committed:
fix(theming): Fix various ci issues (#33147)

msyavuz committed:
Merge branch 'master' into template_less

kgabryje commented on superset/config.py:
```diff
@@ -1623,6 +1614,7 @@ def EMAIL_HEADER_MUTATOR(  # pylint: disable=invalid-name,unused-argument  # noq
 
 # Do you want Talisman enabled?
 TALISMAN_ENABLED = utils.cast_to_boolean(os.environ.get("TALISMAN_ENABLED", True))
+TALISMAN_ENABLED = False
```
Heads up, I'm assuming this shouldnt be here?

kgabryje committed:
Fix changes in DatasourcePanelItem after merge

alexandrusoare committed:
refactor(Tabs): Tabs to use items instead of TabPanel (#33057)

Co-authored-by: Mehmet Salih Yavuz <salih.yavuz@proton.me>

DamianPendrak committed:
fix: upgrade tabs in database modal (#33038)

Co-authored-by: Maxime Beauchemin <maximebeauchemin@gmail.com>
Co-authored-by: Mehmet Salih Yavuz <salih.yavuz@proton.me>

mistercrunch committed:
set colorLink == colorPrimary

msyavuz committed:
fix(Select): Merge conflicts and Select bug (#33165)

msyavuz committed:
Merge branch 'master' into template_less

msyavuz committed:
fix: post merge

EnxDev committed:
refactor(components): Replace native HTML elements with Ant Design v5 components (#33090)

This PR introduces a refactor that replaces native HTML elements with their Ant Design v5 equivalents across the codebase. The goal is to ensure a more consistent UI, better accessibility, and improved maintainability by leveraging standardized components from Ant Design.

msyavuz committed:
refactor(Table): Use Table instead of html <table> in CollectionTable (#33159)

EnxDev committed:
refactor(Components): Use named imports - P.2 (#33249)

msyavuz committed:
fix(Theming): DatabaseModal visual fixes (#33286)

Fix visual regressions in DatabaseModal introduced by theming changes

mistercrunch committed:
fix: button colors and icons (#33287)

EnxDev committed:
Merge branch 'master' into template_less

EnxDev committed:
refactor(InfoTooltipWithTrigger): Replace support for fa icons with antd5 icons (#33256)

- Define and centralized tooltip icon variants (info, warning, notice, error, question)
- Replace old FontAwesome icons
- Update tests to reflect the new icons and variant behavior

mistercrunch committed:
Merge branch 'master' into template_less

mistercrunch commented:
MERGE MERGE MEREG!!!

![git-merge-git](https://github.com/user-attachments/assets/2df735c3-2efd-4e8e-95e1-f529a2615e91)

mistercrunch committed:
fix: various theme-related tweaks and touches (#33372)

EnxDev committed:
Merge branch 'master' into template_less

EnxDev committed:
fix: replace invalid theme tokens

mistercrunch committed:
Merge branch 'master' into template_less

mistercrunch committed:
fix category icon in VizPicker

mistercrunch committed:
fix DnD options caret and separator

mistercrunch committed:
bigger icon in sql lab tab headers

mistercrunch committed:
apply default theme boxShadow to cards

EnxDev committed:
refactor(ui): replace native HTML elements with Ant Design 5 components for consistent theming (#33231)

msyavuz committed:
Merge branch 'master' into template_less

msyavuz committed:
fix: minor import issues

msyavuz committed:
fix: post merge fix imports

mistercrunch committed:
Merge branch 'master' into template_less

mistercrunch committed:
more generic error catching

msyavuz committed:
fix(theming): Fix various visual regressions from theming (#33302)

mistercrunch committed:
dummy commit to trigger CI

mistercrunch committed:
set the test for docker-compose-image-tag to only happen on merge to master

mistercrunch committed:
fix the e2e GHA

mistercrunch committed:
another try at fixing the GHA

mistercrunch commented:
@msyavuz just FYI noticed the `.github/workflows/superset-e2e.yml` was not triggering and saw you had touched it probably to fix some issue around uploading artifacts with 2 with the same name. I think the attempted fix broke the GHA so I pushed a fix for it. I think the fix needs to go in `master` too, probably explains some race condition that happens at times there too. I think this race condition got introduced with the APP_ROOT-related PR somewhat recently.

msyavuz committed:
fix(theming): Merge ci issues and new antd4 component to antd5 (#33429)

msyavuz committed:
feat(theming): Support theming on embedding-sdk (#33360)

mistercrunch committed:
chore: rip antd-v4 from the app! (#33411)

mistercrunch committed:
Merge branch 'master' into template_less

mistercrunch committed:
fixing package-lock

mistercrunch committed:
Merge branch 'master' into template_less

msyavuz committed:
fix(theming): Duplicate select values bug (#33558)

mistercrunch committed:
fix: remove all `less` styling! (#33445)

Co-authored-by: Enzo Martellucci <enzomartellucci@gmail.com>
Co-authored-by: Mehmet Salih Yavuz <salih.yavuz@proton.me>

mistercrunch committed:
fix native filters styling

EnxDev committed:
fix(theming): Fix visual regressions from theming P2/3 (#33587)

msyavuz committed:
feat(Login): Migrate login page to frontend (#33255)

mistercrunch committed:
chore: moving reusable components from `src/components` to `packages/superset-ui-core/src/components` (#33593)

mistercrunch committed:
fix Tooltip-related imports/exports

mistercrunch commented:
<html><head></head><body><h2>📦 Bundle Size Improvements - Ant Design v5 Migration</h2>

Asset Type | Before | After | Reduction | Improvement
-- | -- | -- | -- | --
CSS | 712K | 120K | -592K | -83.1% 🎉
JavaScript | 13.0MB | 10.7MB | -2.3MB | -17.7%
Total | 13.7MB | 10.8MB | -2.9MB | -21.2%


<h3>🚀 Performance Impact</h3>
<ul>
<li><strong>~1MB+ less download</strong> for users (with compression)</li>
<li><strong>83% reduction in CSS</strong> means significantly faster style parsing and rendering</li>
<li><strong>Cleaner bundle</strong> with removal of legacy CSS/LESS files and unused Ant Design v4 dependencies</li>
</ul>
<p><em>Results measured using <code>du -h</code> on built assets in <code>/superset/static/assets/</code></em></p></body></html>

mistercrunch committed:
Merge branch 'master' into template_less

mistercrunch committed:
fix type errors

mistercrunch committed:
fix unit test live

EnxDev committed:
chore(CronPicker): Bump to v5.2.0 for full theming support (#33690)

EnxDev committed:
Merge branch 'master' into template_less

EnxDev committed:
refactor(collapse): fix custom Collapse and align related imports post-merge

EnxDev committed:
fix(RoleList): remove duplicate column

EnxDev committed:
Merge branch 'master' into template_less

EnxDev committed:
refactor(components): align related imports post-merge

mistercrunch committed:
Merge branch 'master' into template_less

EnxDev committed:
fix(RTL test): update tab selector to avoid CSS pseudo-class parsing error

EnxDev committed:
fix(webpack): add stream-browserify fallback for Node.js 'stream' module'

EnxDev committed:
fix(deps): align geostyler versions to resolve dependency conflicts

EnxDev committed:
fix(RTL test): adjust ResultsPaneOnDashboard.test.tsx to avoid invalid selectors :focus-visible

EnxDev committed:
fix(tests): fix missing imports in DatasourceEditorCurrency.test and DatasourceEditorRTL.test

EnxDev committed:
refactor(Input Story): enhance type integration

EnxDev committed:
refactor(UserInfo test): wrap render in await act to ensure state updates are flushed before assertions

EnxDev committed:
fix(RTL test): await asyncRender in DataSourceEditor.test to prevent async updates after test end

mistercrunch commented:
@EnxDev  thanks for fixing the build, I think I broke it while merging `master`, I meant to fix it over the weekend but you beat me to it!

msyavuz committed:
Merge branch 'master' into template_less

msyavuz committed:
fix(DatasourceEditor): clear mocks

msyavuz committed:
fix(e2e): increase timeout

EnxDev commented on superset-frontend/src/components/Datasource/DatasourceEditor.test.jsx:
```diff
@@ -135,7 +135,7 @@ describe('DatasourceEditor', () => {
     await userEvent.type(inputCertifiedBy, 'test');
     await userEvent.type(inputCertDetails, 'test');
-  });
+  }, 4000);
```
I think it's not enough.

EnxDev committed:
Merge branch master into template_less

EnxDev committed:
refactor(DataSourceEditor tests): remove unused await and increase timeout for "can modify columns" test

EnxDev committed:
fix(theming): Fix visual regressions from theming P4 (#33703)

msyavuz committed:
refactor(theming): Remove direct imports of supersetTheme (#33770)

msyavuz committed:
feat(theming): implement a ThemeController (#33420)

EnxDev committed:
fix(BaseIcon): Ensure iconSize is parsed correctly when passed as string (#33780)

msyavuz committed:
fix(Table): table colors

EnxDev committed:
chore(storybook): Include the missing path for loading component inside storybook (#33787)

Adds the path for the components we moved into the packages directory `packages/superset-ui-core/src/components`

EnxDev committed:
Merge branch 'master' into template_less

EnxDev committed:
Merge branch 'master' into template_less

EnxDev committed:
fix CI tests

msyavuz committed:
fix(ThemeController): change mode when theme is edited

msyavuz committed:
fix(ThemeController): update themeObject

msyavuz committed:
fix(ThemeController): update theme object with controller and select-editor compat

mistercrunch committed:
fix SubMenu active button text in home page

EnxDev committed:
fix(theming): Fix visual regressions from theming P5 (#33786)

msyavuz committed:
fix(HeaderMenu): remove mocks

msyavuz committed:
fix(TableChart): background colors test

geido reviewed (approved):
It’s been a great experience contributing to this project, and I’m excited to see it come to life!

msyavuz committed:
fix(e2e): containing label

msyavuz committed:
Merge branch 'master' into template_less

EnxDev reviewed (approved):
Contributing to this project has been a rewarding experience. It’s been a pleasure working with you guys! I’m really looking forward to seeing it in action; thanks so much, everyone ❤️

mistercrunch committed:
re-removing enzyme from package.json

EnxDev committed:
chore(Icons): add import for "UserAddOutlined" icon from AntD

EnxDev committed:
fix(RightMenu): center items vertically

mistercrunch committed:
shut pre-commit/pylint up

rusackas reviewed (approved):
This diff is giving my browser a heart attack... but here's the codeowner stamp it needs!

Thanks (and congratulations) to all involved, this is a  HUGE deal, and a _loooong_ time coming.
